packages/components/src/components/hds/advanced-table/index.ts

```diff
@@ -425,9 +444,71 @@ export default class HdsAdvancedTable extends Component<HdsAdvancedTableSignatur
       this._selectAllCheckbox.indeterminate =
         selectedRowsCount > 0 && selectedRowsCount < selectableRowsCount;
       this._isSelectAllCheckboxSelected = this._selectAllCheckbox.checked;
-      this._selectAllCheckbox.dispatchEvent(
-        new Event('toggle', { bubbles: false })
+    }
+  }
+
+  @action
+  didInsertExpandButton(): void {
+    this.setExpandAllState();
+  }
+
+  @action
+  willDestroyExpandButton(): void {
+    this.setExpandAllState();
+  }
+
+  @action
+  setExpandAllState(): void {
+    if (this._expandAllButton && this._element) {
```

Comment:
suggestion: not a future-proof solution, but to make the logic here work, you would need to delay it until the next cycle (you can wrap the content of this if statement in a `next(() => { /*if content hoes here*/ });`). It will require `import { next } from '@ember/runloop';` and likely a `// eslint-disable-next-line ember/no-runloop`, but it should resolve the state issue.

Reply:
Cool, it does fix the issue!